modules/core/src/lib/layer.js

```diff
@@ -551,6 +554,10 @@ export default class Layer extends Component {
   restorePickingColors() {
     const {pickingColors, instancePickingColors} = this.getAttributeManager().attributes;
     const colors = pickingColors || instancePickingColors;
+    // The picking color cache may have been freed and then reallocated. This ensures we read from the currently allocated cache.
+    if (this.internalState.usesPickingColorCache) {
```

Comment:
Add a check here to avoid unnecessary `subarray` calls?

```suggestion
    if (this.internalState.usesPickingColorCache && colors.value.buffer !== pickingColorCache.buffer) {
```